services/QuillLMS/spec/controllers/concerns/diagnostic_reports_spec.rb

```diff
@@ -41,6 +42,24 @@
     end
   end
 
+  describe '#calculate_proficiency_score' do
```

Comment:
Here's a possible refactor that abstracts out the subject leaving a more conventional expectation block.  The key thing here is that we're naming the parameters in the function call so that the context descriptions don't need to specify the values (which can become stale).
  
```
  describe '#calculate_proficiency_score' do
    subject { calculate_proficiency_score(number_correct, number_incorrect) }

    context 'undefined scores' do
      let(:number_correct) { 0 }
      let(:number_incorrect) { 0 }

      it { expect(subject).to eq DiagnosticReports::NOT_PRESENT }
    end

    context 'zero correct' do
      let(:number_correct) { 0 }
      let(:number_incorrect) { 1 }

      it { expect(subject).to eq 0 }
    end

    context 'zero incorrect' do
      let(:number_correct) { 1 }
      let(:number_incorrect) { 0 }

      it { expect(subject).to eq 1 }
    end

    context 'nonzero correct and incorrect' do
      let(:number_correct) { 1 }
      let(:number_incorrect) { 0 }

      it { expect(subject).to eq 0 }
    end
  end
```